tests/lib/SimpleSAML/ConfigurationTest.php

```diff
@@ -97,6 +104,9 @@ public function testHasValueOneOf() {
      * Test SimpleSAML_Configuration::getBaseURL()
      */
     public function testGetBaseURL() {
+
+        // Need to set a configuration file because the SSP Logger attempts to load it.
+        putenv('SIMPLESAMLPHP_CONFIG_DIR=tests/config/no-options');
```

Comment:
This is not necessary. Instead, the next line should read:

```php
$c = SimpleSAML_Configuration::loadFromArray(array(), '[ARRAY]', 'simplesaml');
```

I admit this is quite obscure. The reason for those parameters is that `SimpleSAML_Configuration::getInstance()` gets, but default, an instance of the configuration called `simplesaml`. This is just a name, but it allows us to load multiple configurations at the same time. In any case, this is the default configuration that you get when you invoke the `getInstance()` method without parameters (as most of the code does).

Now, in `SimpleSAML_Configuration::loadFromArray()`, the instance name is the third parameter, and in order to specify it, you need to specify also the second, which is the _location_ of the configuration. In this case, the location defaults to `[ARRAY]` to signal that this configuration was loaded from an array instead of from a file. It doesn't really matter what you set here, the only thing is that you _need_ to set that parameter to something.

Therefore, if you want to load a default configuration in `SimpleSAML_Configuration` from an array, you need to specify those two parameters, because that's what the code will be looking for later.